src/index.js

```diff
@@ -0,0 +1,42 @@
+const BigNumber = require('bignumber.js');
+
+const MoneySafe = Symbol('MoneySafe');
+
+const createCurrency = ({ decimals }) => {
+  const of = (input, value = new BigNumber(String(input))) => {
+    const plus = b => of(value.plus(of(b)));
+    const times = b => of(value.multipliedBy(of(b)));
+    const div = b => of(value.dividedBy(of(b)));
+
+    return Object.assign(plus, {
+      [MoneySafe]: true,
+      constructor: of,
+      map: f => of(f(value)),
+      valueOf: () => value.toNumber(),
+      plus,
+      add: plus,
+      times,
+      multipliedBy: times,
+      div,
+      dividedBy: div,
+      minus: b => of(value.minus(of(b))),
+      toFixed: (digits = decimals) => value.toFixed(digits),
+      toNumber: () => value.toNumber,
+      toString: () => value.toFixed(decimals)
```

Comment:
I don't know if that belongs into this review, or if that should be a feature request: Should we maybe expose `equals`, too? We use a lot of `toString()` for equality checks in the tests, and I can imagine users having to do the same 🤔

Reply:
Please open a feature request.